### ISSUE TITLE
Kokoro linux: BadZipfile("Truncated file header") or Bad CRC-32 when installing pip packages

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#11968]
Introducing flakiness to python build on kokoro

```
Building wheels for collected packages: coverage, httplib2
  Running setup.py bdist_wheel for coverage: started
  Running setup.py bdist_wheel for coverage: finished with status 'done'
  Stored in directory: /tmp/xdg-cache-home/pip/wheels/a6/34/1f/8fd8e30335ea4d6ecc04606372834b1c976c742db2419e7404
Exception:
Traceback (most recent call last):
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/commands/install.py", line 335, in run
    wb.build(autobuilding=True)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/wheel.py", line 837, in build
    session=self.requirement_set.session)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/download.py", line 809, in unpack_url
    unpack_file_url(link, location, download_dir, hashes=hashes)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/download.py", line 715, in unpack_file_url
    unpack_file(from_path, location, content_type, link)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 599, in unpack_file
    flatten=not filename.endswith('.whl')
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 488, in unzip_file
    data = zip.read(name)
  File "/usr/lib/python2.7/zipfile.py", line 935, in read
    return self.open(name, "r", pwd).read()
  File "/usr/lib/python2.7/zipfile.py", line 968, in open
    raise BadZipfile("Truncated file header")
BadZipfile: Truncated file header
Command exited with non-zero status 2
real 429.64
user 376.66
sys 41.52

2017-07-28 07:06:38,847 FAILED: /var/local/git/grpc/tools/run_tests/helper_scripts/build_python.sh [ret=2, pid=983]
```

Instances
https://sponge.corp.google.com/invocation?tab=Kokoro&id=10e6d07f-8c5e-46de-89b6-20c6570bb104&searchFor=

```
Building wheels for collected packages: coverage, httplib2
  Running setup.py bdist_wheel for coverage: started
  Running setup.py bdist_wheel for coverage: finished with status 'done'
  Stored in directory: /tmp/xdg-cache-home/pip/wheels/a6/34/1f/8fd8e30335ea4d6ecc04606372834b1c976c742db2419e7404
Exception:
Traceback (most recent call last):
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/commands/install.py", line 335, in run
    wb.build(autobuilding=True)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/wheel.py", line 837, in build
    session=self.requirement_set.session)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/download.py", line 809, in unpack_url
    unpack_file_url(link, location, download_dir, hashes=hashes)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/download.py", line 715, in unpack_file_url
    unpack_file(from_path, location, content_type, link)
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 599, in unpack_file
    flatten=not filename.endswith('.whl')
  File "/var/local/git/grpc/py27/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 488, in unzip_file
    data = zip.read(name)
  File "/usr/lib/python2.7/zipfile.py", line 935, in read
    return self.open(name, "r", pwd).read()
  File "/usr/lib/python2.7/zipfile.py", line 630, in read
    data = self.read1(n)
  File "/usr/lib/python2.7/zipfile.py", line 692, in read1
    self._update_crc(data, eof=eof)
  File "/usr/lib/python2.7/zipfile.py", line 647, in _update_crc
    raise BadZipfile("Bad CRC-32 for file %r" % self.name)
BadZipfile: Bad CRC-32 for file 'coverage/tracer.so'
Command exited with non-zero status 2
```

https://sponge.corp.google.com/target?id=1e175712-24b9-42c5-8665-4e5fb310bb30&target=github/grpc/aggregate_tests&searchFor=&show=FAILED&sortBy=STATUS

**Comment by jtattermusch** [comment]
CC @matt-kwong  @kpayson64

**Comment by jtattermusch** [comment]
After merging #12014, I saw a similar problem at least once. Looks like another instance of different docker containers fighting over files in the  shared `/tmp/xdg-cache-home` directory.

```
Building wheels for collected packages: configparser, wrapt
  Running setup.py bdist_wheel for configparser: started
  Running setup.py bdist_wheel for configparser: finished with status 'done'
  Stored in directory: /tmp/xdg-cache-home/pip/wheels/1c/bd/b4/277af3f6c40645661b4cd1c21df26aca0f2e1e9714a1d4cda8
  Running setup.py bdist_wheel for wrapt: started
  Running setup.py bdist_wheel for wrapt: finished with status 'done'
  Stored in directory: /tmp/xdg-cache-home/pip/wheels/22/a5/9f/ca577d0e7b917b681fdbb7b36a41a8a9ae76fb7fec146438dc
Exception:
Traceback (most recent call last):
  File "/var/local/git/grpc/python_pylint_venv/local/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/var/local/git/grpc/python_pylint_venv/local/lib/python2.7/site-packages/pip/commands/install.py", line 335, in run
    wb.build(autobuilding=True)
  File "/var/local/git/grpc/python_pylint_venv/local/lib/python2.7/site-packages/pip/wheel.py", line 837, in build
    session=self.requirement_set.session)
  File "/var/local/git/grpc/python_pylint_venv/local/lib/python2.7/site-packages/pip/download.py", line 809, in unpack_url
    unpack_file_url(link, location, download_dir, hashes=hashes)
  File "/var/local/git/grpc/python_pylint_venv/local/lib/python2.7/site-packages/pip/download.py", line 715, in unpack_file_url
    unpack_file(from_path, location, content_type, link)
  File "/var/local/git/grpc/python_pylint_venv/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 599, in unpack_file
    flatten=not filename.endswith('.whl')
  File "/var/local/git/grpc/python_pylint_venv/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 488, in unzip_file
    data = zip.read(name)
  File "/usr/lib/python2.7/zipfile.py", line 935, in read
    return self.open(name, "r", pwd).read()
  File "/usr/lib/python2.7/zipfile.py", line 630, in read
    data = self.read1(n)
  File "/usr/lib/python2.7/zipfile.py", line 692, in read1
    self._update_crc(data, eof=eof)
  File "/usr/lib/python2.7/zipfile.py", line 647, in _update_crc
    raise BadZipfile("Bad CRC-32 for file %r" % self.name)
BadZipfile: Bad CRC-32 for file 'wrapt/decorators.py'
Command exited with non-zero status 2
```

**Comment by jtattermusch** [comment]
Seems fixed by #12034
